diff --git a/node_modules/@surunnuage/strapi-plugin-expo-notifications/admin/src/pages/HomePage/receiver_item.js b/node_modules/@surunnuage/strapi-plugin-expo-notifications/admin/src/pages/HomePage/receiver_item.js
index d1b050f..68fbdb9 100644
--- a/node_modules/@surunnuage/strapi-plugin-expo-notifications/admin/src/pages/HomePage/receiver_item.js
+++ b/node_modules/@surunnuage/strapi-plugin-expo-notifications/admin/src/pages/HomePage/receiver_item.js
@@ -3,8 +3,8 @@ import { Box } from "@strapi/design-system/Box";
 import { BaseCheckbox } from "@strapi/design-system/BaseCheckbox";
 import { Typography } from "@strapi/design-system/Typography";
 
-export default function ReceiverItem({ item, tokens, removeToken, addToken }) {
-  const { label, value } = item;
+export default function ReceiverItem({ item, index, tokens, removeToken, addToken }) {
+  const { value } = item;
   const isIn = () => tokens.includes(value);
   const manageChange = () => {
     if (isIn()) {
@@ -24,7 +24,7 @@ export default function ReceiverItem({ item, tokens, removeToken, addToken }) {
         />
         <div style={{ marginLeft: 6 }}>
           <div style={{ fontWeight: isIn() ? "700" : "400" }}>
-            <Typography variant="omega">{label}</Typography>
+            <Typography variant="omega">{index+1}. {value}</Typography>
           </div>
         </div>
       </div>
diff --git a/node_modules/@surunnuage/strapi-plugin-expo-notifications/admin/src/pages/HomePage/receivers.js b/node_modules/@surunnuage/strapi-plugin-expo-notifications/admin/src/pages/HomePage/receivers.js
index 6326b8b..7c3e5af 100644
--- a/node_modules/@surunnuage/strapi-plugin-expo-notifications/admin/src/pages/HomePage/receivers.js
+++ b/node_modules/@surunnuage/strapi-plugin-expo-notifications/admin/src/pages/HomePage/receivers.js
@@ -61,10 +61,11 @@ export default function Receivers({
             height: 160,
           }}
         >
-          {receivers.map((item) => (
+          {receivers.map((item, index) => (
             <div key={item.label}>
               <ReceiverItem
                 item={item}
+                index={index}
                 tokens={tokens}
                 addToken={addToken}
                 removeToken={removeToken}
diff --git a/node_modules/@surunnuage/strapi-plugin-expo-notifications/server/services/exponotification.js b/node_modules/@surunnuage/strapi-plugin-expo-notifications/server/services/exponotification.js
index eb9c08c..20a2688 100644
--- a/node_modules/@surunnuage/strapi-plugin-expo-notifications/server/services/exponotification.js
+++ b/node_modules/@surunnuage/strapi-plugin-expo-notifications/server/services/exponotification.js
@@ -47,7 +47,7 @@ module.exports = ({ strapi }) => ({
   },
   async recipientsFrom(start) {
     const count = await strapi.entityService.count(
-      "plugin::users-permissions.user"
+      "api::guest.guest"
     );
     const customFieldName = await strapi
       .plugin("expo-notifications")
@@ -55,18 +55,23 @@ module.exports = ({ strapi }) => ({
     let recipients = [];
     if (customFieldName) {
       const rawRecipients = await strapi.entityService.findMany(
-        "plugin::users-permissions.user",
+        "api::guest.guest",
         {
           start: start,
           // limit: 200,
           filters: {
             [customFieldName]: {
-              $notNull: true,
+              $contains: "ExponentPushToken[",
             },
           },
         }
       );
       rawRecipients.forEach((item) => {
+
+        // check for duplicates
+        if (recipients.find((i) => i.expoPushToken === item[customFieldName])) {
+          return;
+        }
         item.expoPushToken = item[customFieldName];
         recipients.push(item);
       });
